src/main/scala/com/fortysevendeg/android/sqlite/metadata/SQLDroidDatabaseMetaData.scala

```diff
@@ -0,0 +1,933 @@
+package com.fortysevendeg.android.sqlite.metadata
+
+import java.sql._
+
+import android.database.{MatrixCursor, MergeCursor}
+import com.fortysevendeg.android.sqlite.metadata.SQLDroidDatabaseMetaData._
+import com.fortysevendeg.android.sqlite.resultset.SQLDroidResultSet
+import com.fortysevendeg.android.sqlite.util.CursorUtils._
+import com.fortysevendeg.android.sqlite.{SQLDroidDriver, WrapperNotSupported, _}
+import org.sqldroid.SQLDroidConnection
+
+import scala.util.{Failure, Success, Try}
+
+class SQLDroidDatabaseMetaData(connection: Connection)
+  extends DatabaseMetaData
+  with SQLiteDatabaseMetaDataSupport
+  with WrapperNotSupported {
+
+  lazy val attributesStatement = connection.prepareStatement(attributes)
+
+  lazy val bestRowIdStatement = connection.prepareStatement(bestRowId)
+
+  lazy val catalogsStatement = connection.prepareStatement(catalogs)
+
+  lazy val columnPrivilegesStatement = connection.prepareStatement(columnPrivileges)
+
+  lazy val procedureColumnsStatement = connection.prepareStatement(procedureColumns)
+
+  lazy val proceduresStatement = connection.prepareStatement(procedures)
+
+  lazy val superTablesStatement = connection.prepareStatement(superTables)
+
+  lazy val superTypesStatement = connection.prepareStatement(superTypes)
+
+  lazy val tablePrivilegesStatement = connection.prepareStatement(tablePrivileges)
+
+  lazy val tableTypesStatement = connection.prepareStatement(tableTypes)
+
+  lazy val typeInfoStatement = connection.prepareStatement(typeInfo)
+
+  lazy val udtsStatement = connection.prepareStatement(udts)
+
+  lazy val versionColumnsStatement = connection.prepareStatement(versionColumns)
+
+  override def getConnection: Connection = connection
+
+  override def getAttributes(
+    catalog: String,
+    schemaPattern: String,
+    typeNamePattern: String,
+    attributeNamePattern: String
+    ): ResultSet = attributesStatement.executeQuery()
+
+  override def getBestRowIdentifier(
+    catalog: String,
+    schema: String,
+    table: String,
+    scope: Int,
+    nullable: Boolean
+    ): ResultSet = bestRowIdStatement.executeQuery()
+
+  override def getCatalogs: ResultSet = catalogsStatement.executeQuery()
+
+  override def getColumnPrivileges(
+    catalog: String,
+    schema: String,
+    table: String,
+    columnNamePattern: String
+    ): ResultSet = columnPrivilegesStatement.executeQuery()
+
+  override def getColumns(
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Method is longer than 50 lines](https://www.codacy.com/app/47deg/scala-sqlite-droid/file/784801318/issues/source?bid=2442814&fileBranchId=2493741#l71)